src/Dialog/Dialog.tsx

```diff
@@ -287,6 +287,21 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
     [onClose],
   )
 
+  React.useEffect(() => {
+    // If the body is already set to overflow: hidden, it likely means
+    // that there is already a modal open. In that case, we should bail
+    // so we don't re-enable scroll after the second dialog is closed.
+    if (document.body.style.overflow === 'hidden') {
+      return
+    }
+
+    document.body.style.overflow = 'hidden'
+
+    return () => {
+      document.body.style.overflow = ''
```

Comment:
Not blocking: should we keep a track of what it was set to before and reset to that, just in case it was scroll or clip instead of the default visible?